src/main/java/org/codice/usng/CoordinateSystemTranslator.java

```diff
@@ -0,0 +1,830 @@
+                1 - this.eccSquared * Math.sin(latRad) * Math.sin(latRad));
+        double T = Math.tan(latRad) * Math.tan(latRad);
+        double C = this.eccPrimeSquared * Math.cos(latRad) * Math.cos(latRad);
+        double A = Math.cos(latRad) * (lonRad - lonOriginRad);
+
+        // Note that the term Mo drops out of the "M" equation, because phi
+        // (latitude crossing the central meridian, lambda0, at the origin of the
+        //  x,y coordinates), is equal to zero for UTM.
+        double M = this.equatorialRadius * (
+                (1 - this.eccSquared / 4 - 3 * (this.eccSquared * this.eccSquared) / 64
+                        - 5 * (this.eccSquared * this.eccSquared * this.eccSquared) / 256) * latRad
+                        - (3 * this.eccSquared / 8 + 3 * this.eccSquared * this.eccSquared / 32
+                        + 45 * this.eccSquared * this.eccSquared * this.eccSquared / 1024)
+                        * Math.sin(2 * latRad) + (15 * this.eccSquared * this.eccSquared / 256
+                        + 45 * this.eccSquared * this.eccSquared * this.eccSquared / 1024)
+                        * Math.sin(4 * latRad)
+                        - (35 * this.eccSquared * this.eccSquared * this.eccSquared / 3072)
+                        * Math.sin(6 * latRad));
+
+        double UTMEasting = (CoordinateSystemTranslator.K0 * N * (A + (1 - T + C) * (A * A * A) / 6
+                + (5 - 18 * T + T * T + 72 * C - 58 * this.eccPrimeSquared) * (A * A * A * A * A)
+                / 120) + CoordinateSystemTranslator.EASTING_OFFSET);
+
+        double UTMNorthing = (CoordinateSystemTranslator.K0 * (M + N * Math.tan(latRad) * ((A * A) / 2
+                + (5 - T + 9 * C + 4 * C * C) * (A * A * A * A) / 24 +
+                (61 - 58 * T + T * T + 600 * C - 330 * this.eccPrimeSquared) * (A * A * A * A * A
+                        * A) / 720)));
+
+        return new UtmCoordinate(zoneNumber, UTMEasting, UTMNorthing);
+    }
+
+    /**
+     * Converts from lat/lon to USNG.
+     *
+     * @param latLonCoordinate the lat/lon coordinate to be converted.
+     * @return a USNG equivalent of latLonCoordinate.
+     */
+    public UsngCoordinate toUsng(final LatLonCoordinate latLonCoordinate) {
+        return this.toUsng(latLonCoordinate, CoordinatePrecision.ONE_METER);
+    }
+
+    /**
+     * Converts from lat/lon to USNG.
+     *
+     * @param latLonCoordinate the lat/lon coordinate to be converted.
+     * @param coordinatePrecision the requested precision of the returned UsngCoordinate.
+     * @return a USNG equivalent of latLonCoordinate.
+     */
+    public UsngCoordinate toUsng(final LatLonCoordinate latLonCoordinate, final CoordinatePrecision coordinatePrecision) {
+        double lat = latLonCoordinate.getLat();
+        double lon = latLonCoordinate.getLon();
+        int precision = coordinatePrecision.getIntValue() + 1;
+
+        // make lon between -180 & 180
+        if (lon < -180) {
+            lon += 360;
```

Comment:
Should we consider doing a modulo in case the coordinate is greater than even +/-360?